docs/cartesian-product.adoc

```diff
@@ -387,23 +387,46 @@ class MyCartesianTestClass {
 }
 ----
 
-==== Requirements for the static factory method
+You can use an argument provider method in nested classes, as long as these are annotated with `@TestInstance(Lifecycle.PER_CLASS)`.
 
-There are multiple requirements the static factory method has to fulfill to qualify:
+[source,java]
+----
+class MyCartesianTestClass {
+	@Nested
+	@TestInstance(Lifecycle.PER_CLASS)
+	MyNestedCartesianTestClass {
+		@CartesianTest
+		@CartesianTest.MethodFactory("provideArguments")
+		void testNeedingArguments(String string, int i) {
+			// passing test code
+		}
+
+		ArgumentSets provideArguments() {
```

Comment:
```suggestion
		// this provider method doesn't have to be static
		ArgumentSets provideArguments() {
```

Reply:
In fact, it can't even be static because it's in a non-static nested class. I've added a comment about that, as well as why the `@TestInstance` annotation is necessary.